src/PatientChecking/PatientChecking/Services/PatientService.cs

```diff
@@ -18,17 +19,56 @@ public PatientService(PatientCheckInContext patientCheckInContext)
             _patientCheckInContext = patientCheckInContext;
         }
 
-        public async Task<PatientDashboard> GetPatientsSummary()
+        public async Task<PagedResult<PatientListViewModel>> GetListPatientPaging(PagingRequest request)
         {
-            var numberOfPatients = await _patientCheckInContext.Patients.ToListAsync();
-            var numberOfPatientsInMonth = await _patientCheckInContext.Appointments.Where(x => x.CheckInDate.Year == DateTime.Now.Year && x.CheckInDate.Month == DateTime.Now.Month)
-                                                                 .GroupBy(p => p.PatientId)
-                                                                 .Select(g => new { g.Key, count = g.Count() }).ToListAsync();
-            return new PatientDashboard()
+            var query = from patient in _patientCheckInContext.Patients
+                        join contact in _patientCheckInContext.Contacts on patient.PatientId equals contact.PatientId
+                        join address in _patientCheckInContext.Addresses on contact.ContactId equals address.ContactId
+                        where address.IsPrimary == true
+                        select new { patient, contact, address };
+
+            if(request.SortSelection == 0)
+            {
+                query = query.OrderBy(i => i.patient.PatientIdentifier);
+            }
+            else if(request.SortSelection == 1)
+            {
+                query = query.OrderBy(i => i.patient.FullName);
+            }
+            else
+            {
+                query = query.OrderBy(i => i.patient.DoB);
+            }
+
+            int totalRow = query.Count();
+            var data = await query.Skip((request.PageIndex - 1) * request.PageSize)
+                .Take(request.PageSize).Select(x => new PatientListViewModel()
+                {
+                    PatientIdentifier = x.patient.PatientIdentifier,
+                    FullName = x.patient.FullName,
+                    DoB = x.patient.DoB.ToString("dd-MM-yyyy"),
+                    Gender = x.patient.Gender == 0 ? "Male" : x.patient.Gender == 1 ? "Female" : "Others",
```

Comment:
Long has update this to Enum, please pull his code.

Reply:
I have pulled his code.